### PR TITLE
tests for ComposeReview integration

### PR DESCRIPTION
**Pull request by rosscreighton**

### CODE REVIEW

File: imports/ui/components/ComposeReview/ComposeReview.jsx [hunk 1]
```diff
@@ -108,4 +112,8 @@ export default connect(state => ({
   invalidRating: state.ComposeReview.invalidRating,
   rating: state.ComposeReview.rating,
   submitSuccess: state.ComposeReview.submitSuccess,
+}), dispatch => ({
```
Comment (sameckmeier) [inline imports/ui/components/ComposeReview/ComposeReview.jsx]: cool so im happy youre into using dispatch to props 👍

File: imports/ui/components/ComposeReview/ComposeReview.tests.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import React from 'react';
+import { chai } from 'meteor/practicalmeteor:chai';
+import { shallow } from 'enzyme';
+import { spy, stub } from 'sinon';
+import { initJSDom } from '../../../lib/testHelpers';
+import { ComposeReview } from './ComposeReview';
+
+initJSDom();
+
+const setup = () => {
```
Comment (sameckmeier) [inline imports/ui/components/ComposeReview/ComposeReview.tests.js]: i like this as a paradigm 👍

File: imports/ui/components/ComposeReview/ComposeReview.tests.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+};
+
+const contentEventStub = { target: { value: 'some review content' } };
+
+describe('<ComposeReview />', function () {
+  it('correctly renders selected rating', function () {
+    const { wrapper, props } = setup();
+    const selectedRatingButton = wrapper
+      .find('.ComposeReview-ratings button')
+      .at(props.rating - 1);
```
Comment (sameckmeier) [inline imports/ui/components/ComposeReview/ComposeReview.tests.js]: we might want to set this to a const named something like selectedRatingIndex

File: imports/ui/components/ComposeReview/ComposeReview.tests.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+    chai.expect(selectedRatingButton.hasClass('ComposeReview-selected')).to.equal(true);
+  });
+
+  it('calls setContent when content is changed', function () {
+    const { wrapper, props } = setup();
+
+    chai.expect(props.setContent.called).to.equal(false);
```
Comment (sameckmeier) [inline imports/ui/components/ComposeReview/ComposeReview.tests.js]: do we need these? like i understand why we might want them, but calledOnce makes sure that the watched function is called only once, so if they were called when they shouldnt calledOnce would return false.

**Comment by sameckmeier** [comment]
looks good!

blocking comments:
making sure that spied functions havent been called before simulating/calling them

File: imports/ui/components/ComposeReview/ComposeReview.jsx (hunk 1 above)
Comment (rosscreighton) [inline imports/ui/components/ComposeReview/ComposeReview.jsx]: I definitely am, but that wasn't what prompted this change. 

If you don't use map dispatch to props, you have to stub es6 imports in your tests. Stubbing props is MUCH easier than stubbing file imports.

File: imports/ui/components/ComposeReview/ComposeReview.tests.js (hunk 2 above)
Comment (rosscreighton) [inline imports/ui/components/ComposeReview/ComposeReview.tests.js]: Props to redux docs

File: imports/ui/components/ComposeReview/ComposeReview.tests.js (hunk 3 above)
Comment (rosscreighton) [inline imports/ui/components/ComposeReview/ComposeReview.tests.js]: Good call

File: imports/ui/components/ComposeReview/ComposeReview.tests.js (hunk 4 above)
Comment (rosscreighton) [inline imports/ui/components/ComposeReview/ComposeReview.tests.js]: I see your point, but my thinking was that this ensures we don't get a false positive. Let's say the component mounts and  the setContent prop is called for some reason when it shouldn't be, then we change the text in the textarea and setContent is not called like it should be. Now we have a false positive.

I'm definitely open to the idea that I'm over engineering this though

**Comment by rosscreighton** [comment]
@sameckmeier  addressed your comments, merging

**Pull request merged**